### PR TITLE
 React Advanced Data Management

### PR DESCRIPTION
**Pull request by niksseif**
### Changes Requested in Issue:

- Current lesson presents useState() and useEffect() in the context of how to use them instead of the way students had been doing it using class components.
- Need to update this lesson so that it is not relying on the context of class components.
- This lesson should introduce students to:
  - The useEffect() hook
  - How to make network requests in their React applications
  - Conditional rendering
  - Create repo (or branch) and activity for students to be able to implement useEffect, fetch and conditional rendering. The current repo/activity linked in this lesson is being used in the class components lesson. Students will be refactoring the class components into functional/hooks in that lesson
  
 ### Changes Made:
- [x] Updated the lesson to remove reliance on class components and instead focus on using functional components with hooks like useEffect().
- [x] Introduced the useEffect() hook and explained its purpose and usage.
- [x] Added a section on making network requests in React applications, demonstrating how to fetch data.
- [x] Included a section on conditional rendering, showcasing how to conditionally display components based on certain conditions.
- [x]Created a new [repository](https://github.com/turingschool-examples/advanced-data-management-hooks-fe/tree/main) for students to implement the concepts of useEffect, fetch, and conditional rendering.
- [x] This lesson will be partially cover the React lifecycle method "deprecated Advanced Data Management" and the "Intermediate React hooks" lesson.

Please review the changes and let me know if there's anything else that needs to be addressed.

### CODE REVIEW

**Comment by niksseif** [comment]
Missing pieces to add to this lesson :

1. Exercise repo

File: lessons/module-3/lifecycle-methods.md
Comment (letakeane) [inline lessons/module-3/lifecycle-methods.md:1]: Is the point of this new file to just store previous lesson information in order to archive it?

File: lessons/module-3/react-3-advanced-data-management.md [hunk 1]
```diff
@@ -1,179 +1,45 @@
...
-length: 180
-tags: react, ideabox, lifecycle methods, conditional rendering
+title: React 3 - Advanced Data Management.
+length: 3 hours
 module: 3
+tags: react, hooks, useEffect, Network request 
 ---
 
 ## Learning Goals
+* Understand how to perform network request and use it within `useEffect`.
```
Comment (letakeane) [inline lessons/module-3/react-3-advanced-data-management.md]: Maybe reword this to be:
"Be able to perform network requests and make use of the response data with the `useEffect` hook"

File: lessons/module-3/react-3-advanced-data-management.md [hunk 2]
```diff
@@ -1,179 +1,45 @@
...
+* Understand how to perform side effect behavior in a functional component with `useEffect`.
+* Understand how to perform conditional rendering in functional component.
```
Comment (letakeane) [inline lessons/module-3/react-3-advanced-data-management.md]: Maybe reword this to be:
"Be able to conditionally render different views in a function component" 

or something??

File: lessons/module-3/react-3-advanced-data-management.md [hunk 3]
```diff
@@ -1,179 +1,45 @@
...
 
-* Understand the component lifecycle
-* Understand class component lifecycle methods
-* Implement conditional rendering
 
-## Vocabulary
+## Vocab
 
-- `lifecycle method` a set of methods found in the parent class Component that fire at different points during the component lifecycle
+* `useEffect` - A React Hook which enables us to add side effect behavior to a functional component.
```
Comment (letakeane) [inline lessons/module-3/react-3-advanced-data-management.md:16]: Could be nice to add another vocab word: side effect

Let's define what we mean when we say "side effect"!

File: lessons/module-3/react-3-advanced-data-management.md [hunk 4]
```diff
@@ -208,7 +74,7 @@ Once you are set up, you can visit `http://localhost:3001/api/v1/ideas` and you
   </section>
   <section class="answer">
 ### What does `.then()` do? What is the method called on? What does it return?
-`.then()` is a promise prototype method that runs when the promise object it is chained to successfully resolves. It returns a new promise object.
+`.then()` is a method of the Promise prototype that is called on a promise object. It is used to handle the successful resolution of a promise. When the promise resolves, the code inside the `.then()` block is executed. The `.then(`) method itself returns a new promise object, allowing for chaining multiple asynchronous operations together.
```
Comment (letakeane) [inline lessons/module-3/react-3-advanced-data-management.md]: "The `.then(`) method itself "
should be
"The `.then()` method itself "

File: lessons/module-3/react-3-advanced-data-management.md [hunk 5]
```diff
@@ -226,121 +91,188 @@ Pair up and work together on implementing the functionality to `fetch` our ideas
...
-// App.js
+### The Component Lifecycle
 
-this.state = {
-  ideas: [],
-  error: ''
-};
+Every component you create goes through several phases of existing:
+
+* **Mounting:** the component is being assessed, created and rendered on the DOM
```
Comment (letakeane) [inline lessons/module-3/react-3-advanced-data-management.md]: Might add clarification:

Every time the component updates, is it also re-mounting? Or does the "mounting" lifecycle only happen once, the first time the component is rendered?

**Review by letakeane — COMMENTED** [review]
This is great overall! I found just a few areas to tweak the sentences a bit for clarity or typos :)

File: lessons/module-3/lifecycle-methods.md
Comment (niksseif) [reply lessons/module-3/lifecycle-methods.md:1]: Well, kind of. We use the name of this file for the new lesson we created, as our lessons follow the format React1, React2, React3. Since we are not specifically teaching lifecycle methods in this new lesson, but we do cover a part of the lifecycle, we thought it would be appropriate to rename this file. However, we still want to keep it accessible in case students have more questions about the lifecycle, so they can refer to this lesson.

**Comment by niksseif** [comment]
Thanks Leta for your suggestions - Fixed them all.
